# 1.
![[Pasted image 20250316094833.png]]
![[Pasted image 20250316094855.png]]
Web-P:
![[Pasted image 20250316094942.png]]
![[Pasted image 20250316095140.png]]
変更:
![[Pasted image 20250316095030.png]]
![[Pasted image 20250316095041.png]]
![[Pasted image 20250316095205.png]]
背景は、サイトにログインするIDを紐づけることで、ログイン方法には:認証時に各LDAPサーバに利用者IDとパスワードを利用するという記述があるので、各サイトのwebサーバは他2サイトのLDAPサーバに接続必要があるとわかる。
そしてネットワーク図を見ると、各サイトのネットワークではFWがLDAPサーバとwebサーバの間に位置している、webサーバはは各ネットワークにおいてDMZに設置されているようなレイアウトになる。
以上の情報をまとめると、web-Pがアクセス必要なのは、まず自分のLDAPサーバ(LDAP-P)、次に残り２社のLDAPサーバ、それぞれすべてFWに追加する必要があり、プロトコルはLDAP

# 2.
![[Pasted image 20250316100707.png]]
![[Pasted image 20250316100828.png]]
本人の同意を得ずに、許可を得た範囲以外の目的で個人情報を取り扱うこと。

# 3.
![[Pasted image 20250316101008.png]]
![[Pasted image 20250316101039.png]]
![[Pasted image 20250316101525.png]]
![[Pasted image 20250316102014.png]]
c:idPairはclass AccountLinkのinstanceなので、AccountLinkの定義上にある、attribute　childIDはリンク先のIDを持つので、このattributeが間違って利用者ID入力される可能性がある。
d:呼び出されると指定しているので、何かのメソッドかfunctionのことを指していると思われる、ここは複雑に考えすぎずに問題通り見ていくと、誤ったIDを入力した後次の動作を聞いているだけなので、次に呼び出されるidPair.checkChildが回答だとわかる。
![[Pasted image 20250316102738.png]]
![[Pasted image 20250316102953.png]]
![[Pasted image 20250316103031.png]]
AccountLinkでは、3回までIDのパスワードの組み合わせ検証をトライするので、再試行されるということは試行して例外が発生したということなので、元のコードを見る例外発生する可能性のある場所はsiteQのログイントライの部分のみ。
==特に注意すべき点==:idPairのリトライポリシーはtryの回数だけカウントしている、ということはID検証が異常終了した場合だけが対象になる、checkChildの下の記述見れば、ID検証が失敗（ID,パスワード不一致）した場合は、例外ではなく、プログラムが正常に終了しているので、リトライの対象になるコードがnameExceptionが発生する可能性のある26行あたりに限定できる
![[Pasted image 20250316103839.png]]
![[Pasted image 20250316103941.png]]
idPairがfor loopの前に先にresultをNO_ERRORに定義しているので、ログインが繰り返し失敗した場合では、リトライ制限回数を超えるとそのままNO_ERRORの状態で次へ進み、紐付けしてしまうことが発生する

# 4.
![[Pasted image 20250316102712.png]]
![[Pasted image 20250316104219.png]]
![[Pasted image 20250316104234.png]]

# 5.
![[Pasted image 20250316104408.png]]
![[Pasted image 20250316104433.png]]
![[Pasted image 20250316105007.png]]
方法はこれ限りではないが、問題のコードの下にはidPaird.childCheckedの状態をチェックする記述があるので、NameExceptionをthrowする前に、childCheckedをFalseにすれば、問題がある場合ではここでキャッチされる。
